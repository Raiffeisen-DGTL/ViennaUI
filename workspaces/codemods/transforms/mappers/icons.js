export const iconsMapper = {
    ArrowRight: 'Forward',
    Exclamation: 'WarningRing',
    Down: 'SelectOpenDown',
    ArrowLeftRing: 'BackRing',
    Close: 'Close',
    Invisible: 'EyeClosed',
    ArrowUp: 'Up',
    CheckSign: 'SealCheck',
    CheckMark: 'Checkmark',
    CheckMarkRing: 'CheckmarkRing',
    Infinity: 'Infinity',
    Attention: 'WarningTr',
    Exit: 'Exit',
    ArrowsDiagonal: 'ArrowsDiag',
    Burger: 'BurgerHor',
    Favorite: 'FavEmpty',
    ArrowDownRing: 'DownRing',
    ArrowDown: 'Down',
    Buttons: 'KeyboardSettings',
    ArrowsLeftRightBlackRgb: 'ArrowsLeftRight',
    Filter: 'Filter',
    ArrowsUpDownBlackRgb: 'ArrowsTopDown',
    Download: 'Download',
    CircleCheckedDashed: 'CheckmarkPart',
    InfoRing: 'InfoRing',
    CloseRing: 'ClearRing',
    Left: 'GoLeft',
    Compass: 'Safari',
    LeftArrow: 'Back',
    CompassArrow: 'Location',
    ArrowRightRing: 'ForwardRing',
    Coordinates1: 'Coordinates',
    ArrowUpRing: 'UpRing',
    Copy: 'Copy',
    DragAndDrop: 'Drag1',
    Draggable: 'Drag',
    Enter: 'Enter',
    Pin2: 'TerminalOnMap',
    Stop: 'Cancel',
    Plus: 'Add',
    SubmitRefillLink: 'Share',
    OtherRing: 'CurrentRing',
    Mirror: 'Mirror1',
    Queue: 'InQueue',
    Up: 'SelectHide',
    PlusRing: 'AddRing',
    Trash: 'Trash',
    Lock: 'Locked',
    Other: 'MoreHor',
    Settings1: 'Settings',
    Sorting: 'Sorting',
    LogoArrow: 'Pin1',
    More: 'MoreVert',
    Refresh3: 'RefreshRepeat',
    ViewCards: 'Grid',
    Refresh2: 'RefreshRepeat',
    ViewList: 'BurgerHor',
    Refresh: 'Retry1',
    QuestionFill: 'HintFilled',
    Rewind: 'RewindLeft',
    InfoFill: 'InfoFilled',
    RewindRight2: 'CollapseRight',
    RepeatWithErrors: 'RetryWithErrors',
    RewindLeft2: 'CollapseLeft',
    Unlock: 'Unlocked',
    RewindRight: 'RewindRight',
    Question: 'HintRing',
    Right: 'GoRight',
    Save: 'Save1',
    Visible: 'EyeOpened',
    InQueue: 'ListOpen',
    InQueueBack: 'ListClose',
    Unarchive: 'Unarchive',
    Diagram: 'ChartPie',
    Schedule: 'ChartBar2',
    Schedule3: 'ChartLineGrow1',
    ScheduleGrowth: 'ChartBarGrow2',
    Scheme: 'Scheme1',
    Scheme2: 'Scheme2',
    Accounts: 'Docs',
    CheckMagnifier: 'InvoiceSearch',
    Task: 'TaskDone',
    ByReceiptNumber: 'DocNmb',
    ClipBlack: 'Attach',
    DocumentSignAndSend: 'DocSignAndSend',
    DocumentSignVisa: 'DocSignVisa',
    AccountError: 'DocError',
    Document2: 'Clipboard',
    Document3: 'Task',
    Document3Arrow: 'TaskSend',
    DocumentFavorites: 'DocFav',
    Envelope: 'Mail',
    Excel: 'Xls',
    Incoming: 'Mail',
    BetweenYourAccounts: 'AccOperations',
    NewDocument: 'DocNew',
    BriefcaseInterest: 'BriefcaseInterest',
    OfficeSupplies: 'OfficeSupplies',
    Organizations: 'Case',
    Passport: 'Passport',
    Pen: 'Pen',
    Pencil: 'Edit',
    SignChecked: 'DocSignChecked',
    SaveAsTemplate: 'TemplateSave',
    CheckSeal: 'InvoiceSeal',
    Template: 'Template',
    Check: 'Invoice2',
    CurrentAccount: 'DocVerified',
    Seal: 'SealStamp',
    SignPartlySigned: 'PaymentPartlySigned',
    ToYourAccount: 'DocRecieve',
    Check2: 'Invoice1',
    DictionaryAdd: 'AddToDic',
    SignRemove: 'DocSignRemove',
    Schedule2: 'ChartBar1',
    VerifiedTemplate: 'TemplateSaved',
    CheckInfo: 'InvoiceInfo',
    Document: 'Document',
    SignSigned: 'PaymentSigned2',
    DocumentCopy: 'Copy',
    TargetAccount: 'AccTarget',
    DocumentEdit: 'DocEdit',
    DocumentHistory: 'DocHistory',
    SavingsAccount: 'Savings',
    DocumentSeal: 'DocSeal',
    DocumentReissue: 'DocReissue',
    EditBlackRgb: 'Edit',
    DocumentSend: 'DocSend',
    CashWithdrawal: 'AtmCash1',
    Terminal4: 'Terminal',
    Transfer: 'CardToOtherBank',
    Something: 'CardRaif',
    Card: 'CardLock',
    TransferToCard: 'C2c',
    Terminal: 'Acquiring1',
    Cards: 'Cards',
    CardLock: 'CardLock',
    CardHeart: 'CardHeart',
    Open: 'CardAddNew',
    Terminal2: 'Acquiring2',
    Mastercard: 'CardMaster',
    Terminal3: 'Pos',
    Atm: 'AtmCash1',
    Atmcard: 'AtmCards1',
    Bridge: 'Bridge',
    ByCardNumber: 'CardNmb',
    Card1: 'Card',
    CardPinChange: 'CardChangePin',
    DepositFunds: 'DepositOperations',
    HandMoney: 'HandCash1',
    Contribution: 'MoneyBag',
    BundleOfMoney: 'BundleOfMoney',
    Pif: 'Pif',
    RubleTurnover: 'RoubleReturn',
    Calculator: 'Calculator',
    CurrencyEur: 'Euro',
    Euro: 'Euro',
    RubleRadial: 'RoubleRing',
    Exchange: 'Exchange2',
    ToPay: 'ToPay',
    HandMoney2: 'HandCash2',
    DebtFcpp: 'DebtFcpp',
    MoneyBag: 'MoneyBag',
    Ruble: 'Rouble',
    Dollar: 'DollarUs',
    Pay: 'Wallet2',
    RubleArrow: 'RoubleSend',
    CashTransactions: 'CashTransactions',
    PaymentOnReceipt1: 'DocRouble',
    CirculationOfMoney: 'RoubleCashflow',
    GearWheelRuble: 'RoubleGear',
    PlusMinus: 'PlusMinus',
    Credit: 'CreditCash',
    HandCard: 'CardGet',
    Portfolio: 'Case',
    CurrencyCalculator: 'CalculatorCurrency2',
    Purse2: 'Wallet1',
    Interest: 'Percent',
    Pear: 'Pear',
    Coffee: 'Coffee',
    Man2: 'Man',
    ServicesAndService: 'PersonService',
    People: 'People2',
    Furniture: 'Furniture',
    Cocktail: 'Cocktail',
    Cake: 'Cake',
    Grandfather: 'Grandfather',
    SmileNoMouth: 'FaceNoMouth',
    Candles: 'FireHouse',
    Flower: 'Flower',
    Consultation: 'Consultation',
    Tray: 'FoodService',
    LampHome: 'Lamp',
    SmileSad: 'FaceSad',
    BabyCare: 'People1',
    Stroller: 'Stroller',
    IceCream: 'IceCream',
    Champagne: 'GlassesStemware',
    Ambulance: 'Ambulance',
    Chair: 'Chair',
    Key: 'DocSign',
    TransferToAPrivatePerson: 'ToPerson',
    Girl: 'Girl',
    Window: 'Window',
    Cafe: 'Cafe',
    Man: 'Man',
    Smile: 'FaceSmile',
    Woman: 'Women',
    Grandmother: 'Grandmother',
    ManButterfly: 'ManButterfly',
    CupSport: 'Rating',
    Heart: 'Heart',
    ManRing: 'ManRing',
    SmileNormal: 'FaceNeutral',
    Candy: 'Candy',
    Cupboard: 'Cupboard',
    HeartBroken: 'HeartBroken',
    Men: 'People1',
    Cap: 'Cook',
    Fireplace: 'Fireplace',
    Monitor: 'Monitor',
    PoliceCap: 'PoliceCap1',
    Shield2: 'Shield',
    Policeman: 'ArmyCap',
    Libra: 'Libra',
    Deposit: 'Safe',
    Bell: 'Notification',
    Flasher: 'Flasher',
    PoliceCar: 'PoliceCar1',
    Fingerprint: 'Prints2',
    Court: 'Court',
    Shield: 'Insurance',
    ShieldCheck: 'ShieldCheck',
    Design: 'Wacom',
    PostServices: 'Frame',
    Theater: 'TheaterOut',
    Violin: 'Violin',
    Bust: 'Bust',
    Literature: 'Literature',
    Movie: 'Movie',
    Art: 'Music1',
    Picture: 'Picture1',
    Brush: 'Brush',
    Barrel: 'Barrel',
    Box: 'Archive',
    Canister: 'Canister',
    ConcreteMixer: 'ConcreteMixer',
    Equipment: 'Equipment',
    Excavator: 'Excavator',
    Factory: 'Factory',
    Gas: 'Gas',
    Gears: 'Gears',
    Hammers: 'Repair1',
    Lamp: 'LightBulbLamp',
    Oil: 'Oil',
    Pipes: 'Pipes',
    Screw: 'Screw',
    Tower: 'Tower',
    Tractor: 'Tractor',
    Truck: 'Truck',
    Water: 'Water',
    WateringCan: 'WateringCan',
    Wheelbarrow: 'Wheelbarrow',
    Satellite: 'Satellite3',
    Chat: 'Chat2',
    Print: 'Printer',
    Keyboard: 'Keyboard',
    BabbleQuestion: 'ChatAsk',
    SendToMail: 'MailIn',
    Support: 'Headphones1',
    Sms: 'Chat1',
    Sound: 'Sound',
    PhoneInfo: 'MobileInfo',
    QuestionLogo: 'ChatAskRaif',
    SatelliteDish: 'Satellite2',
    Entertainment: 'Console',
    Laptop: 'Laptop',
    CommunicationServices: 'MobileConnecting',
    Headphones: 'Headphones2',
    Bobbin: 'Cinema',
    Camera: 'CameraPhoto',
    Camera1: 'CameraVideo ',
    Babble: 'ChatGroup',
    BabbleLogo: 'ChatRaif',
    Mail: 'MailOut',
    Phone: 'Phone',
    PhoneHome: 'Telephone',
    Microphone: 'Microphone',
    AlcoholTobacco: 'AlcoholAndTobacco',
    Packages: 'OtherGoods',
    PriceTag: 'TagPrice3',
    Nameplate: 'NamePlate',
    BabyClothes: 'ClothesBaby',
    Basket2: 'Grocery2',
    Cashbox: 'Cashbox',
    CarDealers: 'CarDealers',
    Purchases: 'Purchases',
    RentACar: 'CarRent',
    QrCode: 'CodeQr',
    CodeRing: 'CodeBarRing',
    Code: 'CodeBar 2',
    Kids: 'Kids',
    HomeAndAppliances: 'HomeEquipment',
    Footwear: 'Footwear2',
    Beauty: 'Mirror1',
    Clothes: 'Clothes2',
    ClothesShoesAccessories: 'Clothes1',
    Cosmetics: 'Cosmetics2',
    GasStation: 'GasStation',
    HealthAndBeauty: 'Apple',
    Offer: 'Offer',
    PriceTagRuble: 'TagPrice1',
    Products: 'Basket',
    Treatment: 'Treatment2',
    Pharmacies: 'Pharmacy',
    Moon: 'Moon',
    Rain: 'Rain',
    PetSupplies: 'Pet',
    Garden: 'Garden',
    Dolphin: 'Dolphin',
    Cloud: 'Bear',
    Snow: 'Snow',
    Tree: 'Tree',
    Fish: 'Fish',
    Leaf: 'Leaf',
    Sun: 'Sun',
    SunCloud: 'SunCloud',
    Bear: 'Bear1',
    Umbrella2: 'UmbrellaClosed',
    Cow: 'Bull',
    Animal: 'Cat',
    Umbrella: 'UmbrellaOpened',
    Whale: 'Whale',
    Ufo: 'Ufo',
    Electronics: 'Monitor',
    DutyFree: 'CreditConsumer',
    Timer: 'Timer',
    Balloon: 'Balloon',
    Backpack2: 'Suitcase',
    Cup: 'Cup',
    Tent: 'Tent',
    TennisBall: 'TennisBall',
    Skates: 'Skates',
    Fishing: 'Fishing',
    Chess: 'Chess',
    Archer: 'Archer',
    PalmBlack: 'Palm',
    Backpack: 'Backpack',
    Barbell: 'Barbell',
    Bowling: 'Bowling',
    BoxSport: 'Boxing',
    ChaiseLounge: 'Sunshade',
    Darts: 'Darts',
    Drone: 'Drone',
    Flag: 'Flag',
    Goals: 'Goals',
    Golf: 'Golf',
    Leisure: 'Tennis',
    LeisureAndTravel: 'Paris',
    Map: 'MapSearch',
    Medal: 'Medal',
    Mountains: 'Mountains',
    Parachute: 'Parachute',
    PingPong: 'PingPong',
    Robot: 'Robot',
    Skiing: 'Skiing',
    Sport: 'Basketball',
    Tourism: 'Navigation',
    Toys: 'Bear',
    TrainingApparatus: 'Gym',
    Volleyball: 'Volleyball',
    Wave2: 'Boat',
    Wave: 'Wave',
    BankHome: 'BankAndHome',
    HouseWap: 'HouseWap',
    BankIntroduction: 'BranchIn',
    Building: 'Company',
    Bank: 'Branch',
    Castle: 'Castle',
    Crane: 'Crane1',
    Crane2: 'Crane2',
    Fence: 'Fence',
    Heating: 'Heating2',
    Home: 'Home',
    Hotels: 'Hotels',
    Houses: 'City',
    Houses2: 'Houses',
    Repairs: 'Repair2',
    ToBudget: 'ToBudget',
    GeneralEducation: 'Book',
    ArtAndLiterature: 'ArtAndLiterature',
    Binoculars: 'Binoculars1',
    Binoculars2: 'Binoculars2',
    BooksShelf: 'BooksShelf',
    Courses: 'Courses',
    Education: 'Education',
    Glasses: 'GlassesEye',
    Globe: 'Globe2',
    Magnifier: 'Search',
    Microscope: 'Microscope',
    ProfessionalEducation: 'Repair1',
    WorldCopy: 'World',
    Metro: 'Metro',
    Gbdd: 'Fines',
    Cruises: 'Ship',
    Auto: 'AutotransportFront',
    Rocket: 'Rocket',
    Taxi2: 'Taxi',
    Taxi: 'Taxi',
    Speedometer: 'Speedometer',
    MetroTrain: 'Railway2',
    AutoParts: 'CarParts',
    Moto: 'Moto1',
    TrolleyBus: 'TrolleyBus',
    AirTransport: 'Airplane',
    AutoService: 'Wrench',
    Bicycle: 'Moto2',
    Helicopter: 'Helicopter',
    MotorTransport: 'Bus',
    Railway: 'Railway',
    Calendar: 'Calendar',
    Clock: 'History',
    OperationsHistory: 'Retry2',
    Expectation: 'InProgress',
    Accessories: 'Accessories',
    AdEvent: 'CalendarAdd',
    Telegram: 'Telegram',
    Whatsapp: 'Whatsapp',
    Viber: 'PhoneActive',
    Clip: 'Attach',
};
