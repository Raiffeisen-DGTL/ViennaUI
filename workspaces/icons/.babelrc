{
  "presets": ["@babel/env", "@babel/typescript", "@babel/react"],
  "env": {
    "esm": {
      "presets": [["@babel/env", { "modules": false }]]
    }
  }
}
