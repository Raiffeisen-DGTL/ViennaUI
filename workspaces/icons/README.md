# Библиотека иконок

###### Raiffeisenbank Design System

Коллекция иконок для использования в интерфейсах. Иконки представлены в виде React компонентов и оригинальных векторных (`*.svg`) изображений. Изображения напрямую можно использовать из пакета.
