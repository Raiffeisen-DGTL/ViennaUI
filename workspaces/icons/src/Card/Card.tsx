import React, { SVGAttributes } from 'react';

export interface CardProps extends SVGAttributes<SVGElement> {
    [key: string]: any;
    size?: 'xs' | 's' | 'm' | 'l' | 'xl' | number;
    color?: string;
}

const sizes = { xs: 12, s: 16, m: 20, l: 24, xl: 32 };
export const Card: React.FC<CardProps> = (props): JSX.Element => {
    const { color = 'currentColor', size = 'm', ...attrs } = props;
    const d = sizes[size] || size;

    return (
        <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width={d} height={d} fill={color} {...attrs}>
            <defs />
            <path d='M6 14h4.5v-2H6v2z' />
            <path
                fillRule='evenodd'
                d='M5 4a3 3 0 00-3 3v10a3 3 0 003 3h14a3 3 0 003-3V7a3 3 0 00-3-3H5zM4 7a1 1 0 011-1h14a1 1 0 011 1v1H4V7zm0 3h16v7a1 1 0 01-1 1H5a1 1 0 01-1-1v-7z'
                clipRule='evenodd'
            />
        </svg>
    );
};

Card.defaultProps = {
    size: 'm',
    color: 'currentColor',
};

Card.displayName = 'Card';
export default Card;
