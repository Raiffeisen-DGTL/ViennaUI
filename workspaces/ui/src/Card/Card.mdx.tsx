import React from 'react';
import styled from 'styled-components';

const Box = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    line-height: 2rem;

    font-size: 16px;
    height: 100%;
    box-sizing: border-box;
    background-color: #f8f8f8;
    color: #aaabad;
    height: 200px;
`;

export const Demo = (props) => <Box {...props}>Content Area</Box>;
