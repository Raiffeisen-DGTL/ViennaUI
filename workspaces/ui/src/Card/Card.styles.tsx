import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const presets = getPresets('card', {
    base: null,
    header: null,
    footer: null,
    subtitle: null,
    contentTitle: null,
    contentTitleGap: null,

    custom: null,
    customHeader: null,
    customTitle: null,
    customActions: null,
    customContent: null,
    customFooter: null,
});

export const Header = styled.div`
    display: flex;
    ${presets.header}

    ${presets.customHeader}
`;

export const Title = styled.div`
    flex-grow: 1;

    ${presets.customTitle}
`;

export const Actions = styled.div`
    ${presets.customActions}
`;

export const Content = styled.div`
    ${presets.customContent}
`;

export const Footer = styled.div`
    ${presets.footer}

    ${presets.customFooter}
`;

export const Subtitle = styled.div`
    ${presets.subtitle}
`;

export const ContentTitle = styled.div`
    ${presets.contentTitle}

    &:not(:first-child) {
        ${presets.contentTitleGap}
    }
`;

export const Box = styled.div<any>`
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    width: 100%;
    ${presets.base}

    ${({ stretch }) =>
        stretch &&
        css`
            height: 100%;

            ${Content} {
                flex-grow: 1;
            }
        `}

    ${presets.custom}
`;
