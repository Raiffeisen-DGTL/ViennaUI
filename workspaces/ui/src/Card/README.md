# Card

Компонент кароточек.


## Импорт

```
import { Card } from 'vienna-ui';
```

## Свойства / Props

Prop | Type | Default | Description
--- | --- | --- | ---
header | React.ReactNode | false | Контент шапки
title | string | false | Заголовок
actions | React.ReactNode | false | Контент в правой части шапки
footer | React.ReactNode | false | Контент подвала
stretch | boolean | false | Может ли компонент растягиваться под родительский контейнер

## Использование

Компонент принимает в качестве дочерних компонентов любой контент.
Компонент также экспортирует вспомогательные подкомпоненты `Card.Title`, `Card.Subtitle` для создания загловка и подзаголовка карточки.
Для создания заголовком внутри тела карточки экпортируется `Card.ContentTitle`.

```jsx
<Card title='Simple card'>
    <div style={{ width: '600px' }}>Content</div>
</Card>
```

## Контент блока "действия"
##### Свойство `actions`

В `actions` можно передать объект с иконкой, который будет отображён в правом верхнем углу карточки.

```jsx
<Card title='Card with actions' actions={<Settings1 />}>
    Content
</Card>
<Card
    title='Card with actions'
    actions={
        <Groups size='xs'>
            <Settings1 />
            <Close />
        </Groups>
    }>
    Content
</Card>
```

## Шапка
##### Свойство `header`, `title`

Для простых заголовков карточки достаточно передать строку в `title`. Для более сложных используется `header`, который приниает JSX-объект. 

```jsx
<Card
    header={
        <Groups alignItems='baseline'>
            <Card.Title>Card title</Card.Title>
            <Card.Subtitle>Card subtitle</Card.Subtitle>
        </Groups>
    }>
    <Demo />
</Card>
```

#### Многострочный заголовок шапки
##### Компонент `Card.Title`, `Card.Subtitle`

Для создания составных заголовском можно использовать компоненты `Card.Title`, `Card.Subtitle` и компонент `Groups`.

```jsx
<Card
    title='Simple card'
    header={
        <Groups design='vertical' size='s'>
            <Card.Title>Card title</Card.Title>
            <Card.Subtitle>Card subtitle</Card.Subtitle>
        </Groups>
    }>
    <Demo />
</Card>
```

## Подвал
##### Свойство `footer`

```jsx
<Card
    title='Card with footer'
    footer={
        <Groups justifyContent='flex-end'>
            <Button design='outline'>Button example</Button>
            <Button design='accent'>Button example</Button>
        </Groups>
    }>
    <Demo />
</Card>
```

## Заголовок в теле карточки
##### Компонент `Card.ContentTitle`

Компонент также экспортирует подкомпонент `Card.ContentTitle` для заголовков внутри контента.

```jsx
<Card title='Content title' actions={<Settings1 />}>
    <Card.ContentTitle>Section header</Card.ContentTitle>
    <Demo />
    <Card.ContentTitle>Section header</Card.ContentTitle>
    <Demo />
</Card>
```

## Адаптивная высота
##### Свойство `stretch`

Свойство `stretch` позволяет растянуть карточку по высоте до 100% высоты родителя.

```jsx
<Card
    stretch
    title='Simple card'
/>
```
