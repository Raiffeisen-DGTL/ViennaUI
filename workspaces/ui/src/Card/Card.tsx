import React, { useMemo } from 'react';
import { H5 } from '../Typography';
import { WithMargin } from '../Whitespace/utils';
import { Box, Header, Actions, Footer, Content, Title, Subtitle, ContentTitle } from './Card.styles';

interface Props {
    /* Card's content */
    children?: React.ReactNode;

    /* Card's title */
    title?: string;

    /* Header region */
    header?: React.ReactNode;

    /* Region to pass actions icons that will be displayed inside the header */
    actions?: React.ReactNode;

    /* Footer region */
    footer?: React.ReactNode;

    /* Stretch allows to expand card to 100% of parent's height */
    stretch?: boolean;
}

interface HTMLAttributeProps {
    [key: string]: any;
    id?: string;
    className?: string;
    role?: string;
    spellCheck?: boolean;
    tabIndex?: number;
}

export type CardProps = HTMLAttributeProps & Props & WithMargin;

export const Card: React.FC<CardProps> & { Title: any; Subtitle: any; ContentTitle: any } = (props: CardProps) => {
    const { children, title, header, actions, footer, ...attrs } = props;

    const showHeader = header || title || actions;

    const cardTitle = useMemo(() => {
        if (header) {
            return header;
        }

        if (title) {
            return <Card.Title>{title}</Card.Title>;
        }

        return null;
    }, [header, title]);

    return (
        <Box {...attrs}>
            {showHeader && (
                <Header>
                    {<Title>{cardTitle}</Title>}
                    {actions && <Actions>{actions}</Actions>}
                </Header>
            )}
            <Content>{children}</Content>
            {footer && <Footer>{footer}</Footer>}
        </Box>
    );
};

Card.Title = H5;
Card.Subtitle = Subtitle;
Card.ContentTitle = ContentTitle;

Card.displayName = 'Card';
export default Card;
