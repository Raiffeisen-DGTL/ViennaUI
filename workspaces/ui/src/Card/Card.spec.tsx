import React from 'react';
import { Settings } from 'vienna.icons';
import { Card } from './Card';

test('Card', () => {
    const snap = snapshot.render(<Card>Card content</Card>);
    expect(snap).toMatchSnapshot();
});

test('Card w/ titile', () => {
    const snap = snapshot.render(<Card title='Card title'>Card content</Card>);
    expect(snap).toMatchSnapshot();
});

test('Card w/ actions', () => {
    const snap = snapshot.render(
        <Card title='Card title' actions={<Settings />}>
            Card content
        </Card>
    );
    expect(snap).toMatchSnapshot();
});

test('Card w/ header', () => {
    const snap = snapshot.render(
        <Card
            header={
                <>
                    <Card.Title>Card title</Card.Title>
                    <Card.Subtitle>Card subtitle</Card.Subtitle>
                </>
            }>
            Card content
        </Card>
    );
    expect(snap).toMatchSnapshot();
});

test('Card w/ footer', () => {
    const snap = snapshot.render(<Card footer={<>Card footer</>}>Card content</Card>);
    expect(snap).toMatchSnapshot();
});

test('Card w/ content title', () => {
    const snap = snapshot.render(
        <Card>
            <Card.ContentTitle>Section header</Card.ContentTitle>
            <div>Section 1</div>
            <Card.ContentTitle>Section header</Card.ContentTitle>
            <div>Section2</div>
        </Card>
    );
    expect(snap).toMatchSnapshot();
});

test('Card w/ stretch', () => {
    const snap = snapshot.render(
        <Card stretch title='Simple card'>
            Card content
        </Card>
    );
    expect(snap).toMatchSnapshot();
});

test('Card w/ any props', () => {
    const props = { className: 'className', id: 'id', onClick: () => 'Card' };
    const component = <Card {...props}>Card content</Card>;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});
