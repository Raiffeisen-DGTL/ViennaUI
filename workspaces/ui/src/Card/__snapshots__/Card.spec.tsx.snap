// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC"
>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
</div>
`;

exports[`Card w/ actions 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC"
>
  <div
    class="Cardstyles__Header-sc-187vqjx-0 lmKpnW"
  >
    <div
      class="Cardstyles__Title-sc-187vqjx-1 kUbeAs"
    >
      <h5
        class="Headingstyles__Box-sc-16udop9-0 fjHjeT"
        color="brand-primary"
      >
        Card title
      </h5>
    </div>
    <div
      class="Cardstyles__Actions-sc-187vqjx-2 iaiege"
    >
      <svg
        fill="currentColor"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M22 13h-1.312a8.659 8.659 0 01-.664 2.473l1.136.656-1 1.732-1.143-.661a8.816 8.816 0 01-1.817 1.817l.666 1.143-1.732 1-.656-1.136a8.693 8.693 0 01-2.478.665V22h-2v-1.311a8.708 8.708 0 01-2.479-.67l-.655 1.136-1.732-1 .66-1.143A8.811 8.811 0 014.983 17.2l-1.143.666-1-1.732 1.136-.656A8.693 8.693 0 013.311 13H2v-2h1.311c.098-.854.322-1.69.665-2.478L2.84 7.866l1-1.732 1.143.66a8.831 8.831 0 011.811-1.811l-.66-1.143 1.732-1 .655 1.136A8.708 8.708 0 0111 3.311V2h2v1.311a8.693 8.693 0 012.473.665l.656-1.136 1.732 1-.661 1.143a8.836 8.836 0 011.817 1.811l1.143-.66 1 1.732-1.136.656c.343.788.567 1.624.664 2.478H22v2zm-10 6a7.008 7.008 0 007-7 7 7 0 10-7 7zm0-10.9a3.9 3.9 0 110 7.8 3.9 3.9 0 010-7.8zm0 6a2.1 2.1 0 100-4.2 2.1 2.1 0 000 4.2z"
        />
      </svg>
    </div>
  </div>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
</div>
`;

exports[`Card w/ any props 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC className"
  id="id"
>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
</div>
`;

exports[`Card w/ content title 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC"
>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    <div
      class="Cardstyles__ContentTitle-sc-187vqjx-6 lqFCB"
    >
      Section header
    </div>
    <div>
      Section 1
    </div>
    <div
      class="Cardstyles__ContentTitle-sc-187vqjx-6 lqFCB"
    >
      Section header
    </div>
    <div>
      Section2
    </div>
  </div>
</div>
`;

exports[`Card w/ footer 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC"
>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
  <div
    class="Cardstyles__Footer-sc-187vqjx-4 iFsYOy"
  >
    Card footer
  </div>
</div>
`;

exports[`Card w/ header 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC"
>
  <div
    class="Cardstyles__Header-sc-187vqjx-0 lmKpnW"
  >
    <div
      class="Cardstyles__Title-sc-187vqjx-1 kUbeAs"
    >
      <h5
        class="Headingstyles__Box-sc-16udop9-0 fjHjeT"
        color="brand-primary"
      >
        Card title
      </h5>
      <div
        class="Cardstyles__Subtitle-sc-187vqjx-5 iXsFQk"
      >
        Card subtitle
      </div>
    </div>
  </div>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
</div>
`;

exports[`Card w/ stretch 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 bFPQsB"
>
  <div
    class="Cardstyles__Header-sc-187vqjx-0 lmKpnW"
  >
    <div
      class="Cardstyles__Title-sc-187vqjx-1 kUbeAs"
    >
      <h5
        class="Headingstyles__Box-sc-16udop9-0 fjHjeT"
        color="brand-primary"
      >
        Simple card
      </h5>
    </div>
  </div>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
</div>
`;

exports[`Card w/ titile 1`] = `
<div
  class="Cardstyles__Box-sc-187vqjx-7 jQDfoC"
>
  <div
    class="Cardstyles__Header-sc-187vqjx-0 lmKpnW"
  >
    <div
      class="Cardstyles__Title-sc-187vqjx-1 kUbeAs"
    >
      <h5
        class="Headingstyles__Box-sc-16udop9-0 fjHjeT"
        color="brand-primary"
      >
        Card title
      </h5>
    </div>
  </div>
  <div
    class="Cardstyles__Content-sc-187vqjx-3 ijlvdi"
  >
    Card content
  </div>
</div>
`;
